src/ring_theory/polynomial/hermite.lean

```diff
@@ -50,4 +51,71 @@ begin
   simp only [map_one, mul_one, derivative_one, sub_zero]
 end
 
+section coeff
+
+lemma hermite_coeff_recur_zero (n : ℕ) :
+  coeff (hermite (n + 1)) 0 = -(coeff (hermite n) 1) := by simp [coeff_derivative]
+
+lemma hermite_coeff_recur (n k : ℕ) :
+  coeff (hermite (n + 1)) (k + 1) = coeff (hermite n) k - (k + 2) * (coeff (hermite n) (k + 2)) :=
+begin
+  rw [hermite_succ, coeff_sub, coeff_X_mul, coeff_derivative, mul_comm],
+  norm_cast
+end
+
+lemma hermite_coeff_upper (n k : ℕ) : coeff (hermite n) (n + k + 1) = 0 :=
+begin
+  induction n with n ih generalizing k,
+  { apply coeff_C },
+  {  rw [hermite_coeff_recur,
+      (by linarith : n + 1 + k = n + k + 1),
+      (by linarith : n + k + 1 + 2 = n + (k + 2) + 1),
+      ih k, ih (k + 2), mul_zero, sub_zero] }
+end
+
+lemma hermite_coeff_upper' (n k : ℕ) (hnk : n < k) : coeff (hermite n) k = 0 :=
+begin
+  obtain ⟨k, rfl⟩ := nat.exists_eq_add_of_lt hnk,
+  apply hermite_coeff_upper
+end
```

Comment:
I think you should move the proof of the first lemma into the second, the first one is pretty contrived.